controllers/vgmanager_daemonset.go

```diff
@@ -112,17 +117,36 @@ var (
 )
 
 // newVGManagerDaemonset returns the desired vgmanager daemonset for a given LVMCluster
-func newVGManagerDaemonset(lvmCluster lvmv1alpha1.LVMCluster) appsv1.DaemonSet {
+func newVGManagerDaemonset(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) (appsv1.DaemonSet, error) {
 	// aggregate nodeSelector and tolerations from all deviceClasses
 	nodeSelector, tolerations := extractNodeSelectorAndTolerations(lvmCluster)
 	volumes := []corev1.Volume{LVMDConfVol, DevHostDirVol, UDevHostDirVol, SysHostDirVol}
 	volumeMounts := []corev1.VolumeMount{LVMDConfVolMount, DevHostDirVolMount, UDevHostDirVolMount, SysHostDirVolMount}
 	privileged := true
 	var zero int64 = 0
+
+	// try to get vgmanager image from env and on absence get from running pod
+	// TODO: investigate why storing this env in a variable is failing tests
```

Comment:
Is the test failure still a problem?

Reply:
with current implementation it isn't, it happens when we store env values in a variable and use that here